File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/PersonalNameFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class PersonalNameFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "firstName",
            "lastName",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private PersonalName value;

    public PersonalNameFlattenedDto()
    {
        this(new PersonalName());
    }

    public PersonalNameFlattenedDto(PersonalName value)
    {
        this.value = value;
    }

    public PersonalName toPersonalName()
    {
        return this.value;
    }


    public String getFirstName()
    {
        return this.value.getFirstName();
    }

    public void setFirstName(String firstName)
    {
        this.value.setFirstName(firstName);
    }

    public String getLastName()
    {
        return this.value.getLastName();
    }

    public void setLastName(String lastName)
    {
        this.value.setLastName(lastName);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        PersonalNameFlattenedDto other = (PersonalNameFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/PersonalNameFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\FlattenedDtoInterface;

class PersonalNameFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'firstName' => 'string',
            'lastName' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var PersonalName
     */
    private $value;

    /**
     * @param PersonalName $value
     */
    public function __construct(PersonalName $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new PersonalName();
        }
    }

    /**
     * @return PersonalName
     */
    public function toPersonalName()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getFirstName()
    {
        return $this->value->getFirstName();
    }

    /**
     * @param string $firstName
     */
    public function setFirstName($firstName)
    {
        $this->value->setFirstName($firstName);
    }

    /**
     * @return string
     */
    public function getLastName()
    {
        return $this->value->getLastName();
    }

    /**
     * @param string $lastName
     */
    public function setLastName($lastName)
    {
        $this->value->setLastName($lastName);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getFirstName(),
            $this->getLastName(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return PersonalNameFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setFirstName($pValues[0]);
        $this->setLastName($pValues[1]);
        return $this;
    }

}

